src/queries/__tests__/a11yState.test.tsx

```diff
@@ -251,3 +258,97 @@ test('byA11yState queries support hidden option', () => {
     `"Unable to find an element with expanded state: false"`
   );
 });
+
+test('*ByA11yState deprecation warnings', () => {
+  const mockCalls = (console.warn as ConsoleLogMock).mock.calls;
+  const view = render(<View accessibilityState={{ disabled: true }} />);
+
+  view.getByA11yState({ disabled: true });
+  expect(mockCalls[0][0]).toMatchInlineSnapshot(`
+    "getByA11yState(...) is deprecated and will be removed in the future.
+
+    Use getByRole(role, { disabled, selected, checked, busy, expanded }) query or expect(...).toHaveAccessibilityState(...) matcher instead."
```

Comment:
Is it worth pointing out that the matcher comes from `@testing-library/jest-native`?

Reply:
Added! 👍🏻